app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/profile/ProfileScreen.kt

```diff
@@ -169,3 +184,79 @@ fun AppVersion() {
         )
     }
 }
+
+@Composable
+fun HandleAction(action: ClickAction, viewModel: ProfileViewModel) {
+    when (action) {
+        ClickAction.NAME -> HandleNameClick(viewModel)
+        ClickAction.PREFERRED_LANGUAGE -> TODO()
+        ClickAction.APP_THEME -> TODO()
+        ClickAction.NOTIFICATION_REMINDER -> TODO()
+        ClickAction.RATING_FEEDBACK -> TODO()
+        ClickAction.PRIVACY_POLICY -> TODO()
+    }
+}
+
+@Composable
+fun HandleNameClick(viewModel: ProfileViewModel) {
+    IPTextInputDialog(
+        titleRes = R.string.name,
+        inputText = viewModel.currentPopupValue,
+        onTextUpdate = { viewModel.currentPopupValue = it  },
+        textInputAttributes = TextInputAttributes(
+            labelTextId = R.string.name
+        ),
+        onCancelClick =  { viewModel.setAction(null, "")},
+        onConfirmClick = { viewModel.setName(viewModel.currentPopupValue) }
+    )
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun IPTextInputDialog(
+    @StringRes titleRes: Int,
+    inputText: String,
+    onTextUpdate: (String) -> Unit,
+    textInputAttributes: TextInputAttributes,
+    onCancelClick: () -> Unit,
+    onConfirmClick: () -> Unit
+) {
+    AlertDialog(
+        onDismissRequest = {}) {
+        Surface(
+            shape = Shapes.extraLarge,
+            color = MaterialColorPalette.surfaceContainerHigh,
+            tonalElevation = dimensionResource(id = R.dimen.dimension_6dp),
```

Comment:
do we require this?

Reply:
This specs are same as Compose default Alertdialog, just copied every token from that component